attachment_synchronize/models/task.py

```diff
@@ -127,7 +127,7 @@ def run_import(self):
         backend = self.backend_id
         all_filenames = backend._list(relative_path=self.filepath)
         if self.filename:
-            filenames = [x for x in all_filenames if self.filename in x]
+            filenames = [x for x in all_filenames if fnmatch(x, self.filename)]
```

Comment:
Much better indeed. Actually, previous version, it managed regex (because it was using fs lib that implement regex filter on the file names).

The best solution, would be to remove this and put this logic directly in storage_backend.
Indeed, storage backend is used to interact with external storage and lack this feature. Better to implement directly in storage backend so it may be used in any sub module, like attachment_synchronize.

So, the idea would be to have here something like : 
`all_filenames = backend._list(relative_path=self.filepath, wildcard=self.filename)`

And in storage backend, implement this, I am not sure what is the better way though.
I guess we could do something like this
https://github.com/OCA/storage/blob/12.0/storage_backend/models/storage_backend.py#L45
    def _list(self, relative_path="", wildcard=None):
        names = self._forward("list", relative_path)
        names = self._forward("list_filter", relative_path, names, wildcard)

And here : https://github.com/OCA/storage/blob/12.0/storage_backend/components/base_adapter.py#L10
Define the method list_filter (or any better method name..)
Like 
```
def list_filter(self, names, wildcard):
    if wildcard:
        Manage regex filtering...
    return names
```
What do you think

Reply:
@florian-dacosta sound a good idea to implement in `storage_backend`, question is ..what maintainer think about that? could we get stuck waiting for them to decide what's best?

Reply:
@GSLabIt
I talked to Sebastien Beau who is a maintener of storage and he agrees it is a good idea to implement it there.
Of course, it may take some time to get it merge, but I'll try to press them !
Anyway, once there is a PR on storage, even if it is not accepted yet, we can still merge this and make the PR on OCA depends on the one of storage.

Making stuff be merged into OCA take some time anyway, some if you need this module fast, you should use it before it is merged!

Reply:
@florian-dacosta ok then, let's propose improvement on `storage_backend`

Reply:
probably a modified version or list will be enough

Reply:
Yes, I guess that for the first version at least, we can just modify `_list` method.
If there is need for further hook later (if we don't want this behavior for a special storage backend for example) we'll improve the module then.

About your proposal, I am not regex expert, seems enough to me.
FYI, here how it was implemented in fs lib (that was used in v9 module)
```
        if wildcard is not None:                                                    
            if not callable(wildcard):                                              
                wildcard_re = re.compile(fnmatch.translate(wildcard))               
                wildcard = lambda fn: bool(wildcard_re.match(fn))                   
            entries = [p for p in entries if wildcard(p)]
```

The callable part is probably too much for now, but they use `re` and `fnmatch`, I am not sure why

Reply:
well not sure, s fnmatch should use regex internally, at least that's what i found .. have a look at: https://pymotw.com/2/fnmatch/#filtering. we can even use filter for our purpose ..

Reply:
filter seems fine.
Let's do this, and see what people on storage backend think.

Reply:
@florian-dacosta r u going to make a PR? or should I do?

Reply:
If you can do it, that's great.
I'll review it

Reply:
ok, i'll ping you once done. Thx